application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -215,50 +341,34 @@ private static void handleRoleSelection(final @NotNull SelectionMenuEvent event,
         handleRoleModifications(event, event.getMember(), guild, rolesToAdd, rolesToRemove);
     }
 
+    /**
+     * Creates a function that maps the {@link SelectOption} to a {@link Role} from the given
+     * {@link Guild}.
+     *
+     * @param guild The {@link Guild} to grab the roles from.
+     *
+     * @return A {@link Function} which maps {@link SelectOption} to the relating {@link Role}.
+     */
+    @Contract(pure = true)
     @NotNull
-    private static Function<SelectOption, Role> roleFromSelectOptionFunction(Guild guild) {
-        return selectedOption -> {
+    private static Function<SelectOption, Role> roleFromSelectOptionFunction(final Guild guild) {
+        return (SelectOption selectedOption) -> {
             Role role = guild.getRoleById(selectedOption.getValue());
 
             if (null == role) {
-                handleNullRole(selectedOption);
+                logRemovedRole(selectedOption);
             }
 
             return role;
         };
     }
 
-    /**
-     * Handles the selection of the {@link SelectionMenu} if it came from a builder.
-     *
-     * @param event the <b>unacknowledged</b> {@link ComponentInteraction}
-     * @param selectedRoles the {@link Role roles} selected by the {@link User} from the
-     *        {@link ComponentInteraction} event
-     */
-    private void handleNewRoleBuilderSelection(@NotNull final ComponentInteraction event,
-            final @NotNull Collection<? extends Role> selectedRoles) {
-        SelectionMenu.Builder menu =
-                SelectionMenu.create(generateComponentId(event.getUser().getId()))
-                    .setPlaceholder("Select your roles")
-                    .setMaxValues(selectedRoles.size())
-                    .setMinValues(0);
-
-        selectedRoles.forEach(role -> menu.addOption(role.getName(), role.getId()));
-
-        event.getChannel()
-            .sendMessageEmbeds(event.getMessage().getEmbeds().get(0))
-            .setActionRow(menu.build())
-            .queue();
-
-        event.reply("Message sent successfully!").setEphemeral(true).queue();
-    }
-
     /**
      * Logs that the role of the given {@link SelectOption} doesn't exist anymore.
      *
      * @param selectedOption the {@link SelectOption}
      */
-    private static void handleNullRole(final @NotNull SelectOption selectedOption) {
+    private static void logRemovedRole(final @NotNull SelectOption selectedOption) {
```

Comment:
It also sends something to the user, not just `log`. I would keep the name as `handle`, otherwise it sounds too innocent.

Reply:
![image](https://user-images.githubusercontent.com/49957334/156326142-480cbc53-cc10-4dc3-986a-d18d71a9eb7b.png)


Yep, you got baited \:p
![image](https://user-images.githubusercontent.com/49957334/156326177-60393c5a-5116-4bc1-9fde-174120f05912.png)